src/clr-addons/datagrid/datagrid-state-persistence/state-persistence-key.directive.ts

```diff
@@ -1,35 +1,156 @@
-import { AfterContentInit, ContentChild, Directive, Input } from '@angular/core';
-import { ClrDatagridPagination } from '@clr/angular';
+import { AfterContentInit, ContentChild, ContentChildren, Directive, Input, OnDestroy, QueryList } from '@angular/core';
+import { ClrDatagrid, ClrDatagridFilter, ClrDatagridPagination } from '@clr/angular';
 import { ClrDatagridStatePersistenceModel } from './datagrid-state-persistence-model.interface';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+const DATE_TYPE = 'date';
 
 @Directive({
   selector: '[clrStatePersistenceKey]',
 })
-export class StatePersistenceKeyDirective implements AfterContentInit {
-  @Input()
-  clrStatePersistenceKey: string;
+export class StatePersistenceKeyDirective implements AfterContentInit, OnDestroy {
+  @Input('clrStatePersistenceKey')
```

Comment:
could we skip the breaking change here? by using another input and hardcoding a default value for the `serverDriven` ?

Reply:
I also thought of that, but because of the drawbacks stated in the documentation I want the users to actively decide what they are doing